File: binary_search.py
import random 

def binary_search(sorted_list, target):
    # O(log(n))
    left_ix = 0
    right_ix = len(sorted_list) - 1
    mid_ix = (left_ix +right_ix) // 2
    num = sorted_list[mid_ix]
    while num != target:
        if abs(left_ix-right_ix) < 2:
            return "Number not in list"
        if num < target:
            left_ix = mid_ix
        if num > target:
            right_ix = mid_ix
        mid_ix = (left_ix +right_ix) // 2
        num = sorted_list[mid_ix]
    return sorted_list[mid_ix]




if __name__ == '__main__':
    numbers = [random.randint(0, 1000) for _ in range(200)]
    numbers.sort()

    print(binary_search(numbers, 25))

File: binarySearch/binarySearch/binarySearch.cpp
// binarySearch.cpp : This file contains the 'main' function. Program execution begins and ends there.
//

#include <iostream>
#include <ctime>

int binarySearch(int ordered_list[], int target, int length)
{
	int left_ix = 0;
	int right_ix = length;

	int mid_ix = floor((left_ix + right_ix) / 2);
	while (abs(right_ix - left_ix) >= 2)
	{
		//std::cout << ordered_list[mid_ix] << '\n';
		if (ordered_list[mid_ix] == target)
		{
			return true;
			//return ordered_list[mid_ix];
		}
		if (ordered_list[mid_ix] < target)
		{
			left_ix = mid_ix;
		}
		if (ordered_list[mid_ix] > target)
		{
			right_ix = mid_ix;
		}
		mid_ix = floor((left_ix + right_ix) / 2);
	}
	return false;
}


int main()
{
	const int targ = 3;
	const int n = 5;
	int list[n] = { 1, 3, 5, 7, 9};

	int found = binarySearch(list, targ, n-1);

	if (found == 1)
	{
		std::cout << "Number found" << '\n';
	}
	else 
	{
		std::cout << "Number not found" << '\n';
	}
}

